app/models/username_block.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+# == Schema Information
+#
+# Table name: username_blocks
+#
+#  id                  :bigint(8)        not null, primary key
+#  allow_with_approval :boolean          default(FALSE), not null
+#  exact               :boolean          default(FALSE), not null
+#  username            :string           not null
+#  created_at          :datetime         not null
+#  updated_at          :datetime         not null
+#
+
+class UsernameBlock < ApplicationRecord
+  HOMOGLYPHS = {
+    '1' => 'i',
+    '2' => 'z',
+    '3' => 'e',
+    '4' => 'a',
+    '5' => 's',
+    '7' => 't',
+    '8' => 'b',
+    '9' => 'g',
+    '0' => 'o',
+  }.freeze
```

Comment:
Shouldn't that also apply to the record itself? afaict, with the current code, you can add “1488” and it will not match anything, because candidates will be transformed to “ia88” before comparison.